### ISSUE TITLE
Add PMI and Brown clustering (distributional features)

### ISSUE THREAD

**Issue by craigtrim** [craigtrim/pystylometry#83]
## Summary

Add two distributional / co-occurrence techniques that pystylometry does not currently cover:

- **PMI** (Pointwise Mutual Information) and **NPMI** (Normalized PMI) — bigram/skipgram association statistics, useful for collocation extraction and stylistic fingerprinting beyond raw frequency.
- **Brown clustering** — hierarchical agglomerative word-class induction from bigram statistics.

Placeholder stubs have been committed so the homes are reserved:

- \`pystylometry/ngrams/pmi.py\`
- \`pystylometry/lexical/brown_clusters.py\`

plus two planning READMEs (\`README_PMI.md\`, \`README_BROWN.md\`) next to them. This issue tracks the real implementation.

### PMI → \`pystylometry/ngrams/\`

PMI is a bigram/skipgram association statistic and belongs alongside the existing n-gram work (\`entropy.py\`, \`extended_ngrams.py\`). Not \`lexical/\` (vocabulary richness) and not \`authorship/\` (profile-distance metrics).

Reuse the bigram/skipgram counting already exposed by \`pystylometry/ngrams/extended_ngrams.py\` rather than re-implementing.

### Brown clustering → \`pystylometry/lexical/\`

Although the **input** is n-grams, the **output** is a word-class assignment, which places it squarely next to \`word_class.py\` and \`word_class_distribution.py\`. Same pattern they already follow.

### Rejected alternative

A standalone \`pystylometry/distributional/\` subpackage — it would duplicate infrastructure that \`ngrams/\` and \`lexical/\` already provide, and breaks the \"one coherent metric family per top-level module\" convention.

### PMI — \`pystylometry/ngrams/pmi.py\`

\`\`\`python
def compute_pmi(text: str, window: int = 2, min_count: int = 5) -> PMIResult: ...
def compute_npmi(text: str, window: int = 2, min_count: int = 5) -> PMIResult: ...
def compute_top_collocations(text: str, k: int = 50, measure: str = \"npmi\") -> list[tuple[str, str, float]]: ...
\`\`\`

Export from \`pystylometry/ngrams/__init__.py\`.

### Brown clustering — \`pystylometry/lexical/brown_clusters.py\`

\`\`\`python
def compute_brown_clusters(text: str, num_clusters: int = 100, min_count: int = 5) -> BrownClusterResult: ...
def compute_brown_class_distribution(text: str, clusters: BrownClusterResult) -> BrownClassDistributionResult: ...
\`\`\`

Export from \`pystylometry/lexical/__init__.py\`.

\`compute_brown_clusters\` trains clusters on the input text. \`compute_brown_class_distribution\` applies pre-trained clusters as a stylometric feature — this is the real use case for authorship analysis (train on a reference corpus, apply as a feature vector).

## Result Dataclasses

All result types must live in \`pystylometry/_types.py\` (single source of truth — see CLAUDE.md architecture notes). Add:

- \`PMIResult\` — contains top collocations, full (w1, w2) -> score map, and summary stats.
- \`BrownClusterResult\` — word -> cluster id map, cluster hierarchy, merge history.
- \`BrownClassDistributionResult\` — cluster id -> frequency / proportion, for use as a feature vector.

## Optional-Extras Gating

Both techniques can be implemented in pure stdlib (Counters + dicts) for a reasonable baseline, **no new extras required**.

If performance demands a heavier backend later (scipy.sparse, scikit-learn), add a new extra in \`pyproject.toml\` under \`[tool.poetry.extras]\` and guard imports in \`pystylometry/__init__.py\` with the existing \`try/except ImportError\` + \`_*_AVAILABLE\` pattern — same as \`readability\` / \`prosody\` / \`syntactic\`. Do **not** import heavy deps at the top level of a module that core users might import.

## Tests

Mirror the module layout:

- \`tests/ngrams/test_pmi.py\`
- \`tests/lexical/test_brown_clusters.py\`

Shared fixtures go in \`tests/conftest.py\` / \`tests/fixtures/\`. Cover: basic correctness on small hand-checked inputs, edge cases (single-token text, all-unique bigrams, below-min-count filtering), and at least one realistic paragraph-scale smoke test.

## Cross-Cutting Wiring Checklist

When implementing:

- [ ] Implementations in \`pmi.py\` / \`brown_clusters.py\`, replacing the placeholder \`# TODO\`.
- [ ] Result dataclasses added to \`pystylometry/_types.py\`.
- [ ] Public exports added to \`pystylometry/ngrams/__init__.py\` and \`pystylometry/lexical/__init__.py\`.
- [ ] Reuse bigram/skipgram helpers from \`pystylometry/ngrams/extended_ngrams.py\` where possible.
- [ ] Tests added under \`tests/ngrams/\` and \`tests/lexical/\`.
- [ ] Module rows added to \`pystylometry/README.md\` and top-level \`README.md\` tables.
- [ ] If exposed on the CLI, add \`*_cli\` functions in \`pystylometry/cli.py\` and \`[tool.poetry.scripts]\` entries in \`pyproject.toml\`.
- [ ] Consider whether \`analyze()\` and \`AnalysisResult\` in \`_types.py\` should expose these as toggleable families.
- [ ] Delete the planning READMEs (\`README_PMI.md\`, \`README_BROWN.md\`) once the real module READMEs absorb their content.

## Verification

- \`make test\` — full suite including new test files.
- \`pytest tests/ngrams/test_pmi.py -v\` / \`pytest tests/lexical/test_brown_clusters.py -v\` — targeted runs.
- \`make lint\` — ruff + strict mypy (\`disallow_untyped_defs = true\`); new dataclasses and functions must be fully annotated.
- Smoke: \`from pystylometry.ngrams import compute_pmi; from pystylometry.lexical import compute_brown_clusters\`.
